sanic/server.py

```diff
@@ -154,6 +157,8 @@ def data_received(self, data):
             self.headers = []
             self.parser = HttpRequestParser(self)
 
+        self.state['requests_count'] = self.state['requests_count'] + 1
```

Comment:
1. Request count should be incremented regardless of http parsing exception
2. I am not sure, but incrementing in `data_received` is also kind of problematic I think.